### PR TITLE
[Spot-43][ML] Job count at InvalidDataHandler.scala:43 taking too long.

### PR DESCRIPTION
**Pull request by rabarona**
Changed the place to call InvalidDataHandler.scala for flow suspicious connects analysis. 
This way, the function `showAndSaveCorruptRecords` won't try to filter `scoredFlowRecords` instead it will filter `dataWithWords` in FlowSuspiciousConnectsMode.scala.

Also removing the line from SpotLDAWrapper, documentDictionary.count() since it showed to be taking a long time as well. 

Tested with these changes and time was reduced from 11 hours to 7. 
More about this issue and steps to reproduce in Jira: https://issues.apache.org/jira/browse/SPOT-43

### CODE REVIEW

File: spot-ml/src/main/scala/org/apache/spot/netflow/model/FlowSuspiciousConnectsModel.scala [hunk 1]
```diff
@@ -199,6 +199,16 @@ object FlowSuspiciousConnectsModel {
     val dataWithWords = totalRecords.withColumn(SourceWord, flowWordCreator.srcWordUDF(ModelColumns: _*))
       .withColumn(DestinationWord, flowWordCreator.dstWordUDF(ModelColumns: _*))
 
+    dataWithWords.cache()
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/netflow/model/FlowSuspiciousConnectsModel.scala]: how does this fix the problem?

Comment (rabarona) [inline spot-ml/src/main/scala/org/apache/spot/netflow/model/FlowSuspiciousConnectsModel.scala]: Caching dataWithWords helps performance as it's used in 2 jobs, one to get corrupt records (error on word creation, flagged as word_error) and then it's primary function which is document word count. 
Cached ensures dataWithWords is not re-calculated when it's materialized in SpotLDAWrapper.

**Comment by lujacab** [comment]
+1

**Comment by raypanduro** [comment]
+1

File: spot-ml/src/main/scala/org/apache/spot/netflow/FlowSuspiciousConnectsAnalysis.scala [hunk 2]
```diff
@@ -44,8 +44,6 @@ object FlowSuspiciousConnectsAnalysis {
     val invalidFlowRecords = filterAndSelectInvalidFlowRecords(inputFlowRecords)
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/netflow/FlowSuspiciousConnectsAnalysis.scala]: it feels weird to me that logging of the corrupt/invalid records is happening after the analysis... shouldn't be happening in the front during the cleaning stages?

Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/netflow/FlowSuspiciousConnectsAnalysis.scala]: eg around line 27

Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/netflow/FlowSuspiciousConnectsAnalysis.scala]: no reason to take it up now though

**Comment by NathanSegerlind** [comment]
what kind of testing did you do for this?

**Comment by rabarona** [comment]
I ran the same data set with the changes and checked the time to complete and the place where _save corrupt records_ was called.
@lujacab ran the same data set and checked the running time. 
Passed all unit tests.

**Pull request merged**